src/Validation/Validation.php

```diff
@@ -733,26 +730,38 @@ public static function decimal(mixed $check, int|bool|null $places = null, ?stri
         }
 
         if ($regex === null) {
-            $lnum = '[0-9]+';
-            $dnum = "[0-9]*[\.]{$lnum}";
-            $sign = '[+-]?';
-            $exp = "(?:[eE]{$sign}{$lnum})?";
-
-            if ($places === null) {
-                $regex = "/^{$sign}(?:{$lnum}|{$dnum}){$exp}$/";
-            } elseif ($places === true) {
-                if (is_float($check) && floor($check) === $check) {
-                    $check = sprintf('%.1f', $check);
-                }
-                $regex = "/^{$sign}{$dnum}{$exp}$/";
-            } else {
-                $places = '[0-9]{' . $places . '}';
-                $dnum = "(?:[0-9]*[\.]{$places}|{$lnum}[\.]{$places})";
-                $regex = "/^{$sign}{$dnum}{$exp}$/";
-            }
+            [$check, $regex] = static::getDecimalRegex($check, $places);
+        }
+
+        return static::_check($check, $regex);
+    }
+
+    /**
+     * Checks that a value is a valid decimal. Both the sign and exponent are optional.
+     *
+     * The currently set locale is used to determine the decimal and thousands separator of the given number.
+     *
+     * Valid Places:
+     *
+     * - null => Any number of decimal places, including none. The '.' is not required.
+     * - true => Any number of decimal places greater than 0, or a float|double. The '.' is required.
+     * - 1..N => Exactly that many number of decimal places. The '.' is required.
+     *
+     * @param mixed $check The value the test for decimal.
+     * @param int|true|null $places Decimal places.
+     * @param string|null $regex If a custom regular expression is used, this is the only validation that will occur.
+     * @return bool Success
+     */
+    public static function localizedDecimal(mixed $check, int|bool|null $places = null, ?string $regex = null): bool
+    {
+        if (!is_scalar($check)) {
+            return false;
+        }
+
+        if ($regex === null) {
+            [$check, $regex] = static::getDecimalRegex($check, $places);
         }
 
-        // account for localized floats.
         $locale = ini_get('intl.default_locale') ?: static::DEFAULT_LOCALE;
```

Comment:
Surely we have a better way to get the active locale.

Reply:
`I18n::getLocale()` would be a cleaner looking option which reads the same value, but the `I18n` package is not a hard dependency of the `Validation` package.